### PR TITLE
[rtloader] high-level memory profiling metrics

### PR DESCRIPTION
**Pull request by truthbk**
### What does this PR do?

Adds high-level memory profiling metrics, these should help us figure out if the RTLoader is leaking anywhere, and the amount of memory being used by it at all times.

### Motivation

Better insights into the RTLoader memory allocation code.

### Additional Notes

synchronous code necessary in the memory tracker, but not a particularly hot-path so performance impact should be reasonable. 

### Related PRs

https://github.com/DataDog/datadog-agent-buildimages/pull/10

### CODE REVIEW

**Review by ogaca-dd — COMMENTED** [review]
Everything looks good to me but please check my comment on `MemoryTracker`

File: rtloader/common/memory.h [hunk 1]
```diff
@@ -40,6 +40,13 @@ void *_malloc(size_t sz);
 */
 void _free(void *ptr);
 
+#undef strdup
+#ifdef __cplusplus
+char *strdup(const char *s1) __THROW;
```
[the remark this answers is not in the document]
Comment (truthbk) [reply rtloader/common/memory.h]: I had to add it, or else I got compilation errors on some platforms. I don't believe it's a problem.

**Comment by ogaca-dd** [comment]
Everything looks good to me!

File: pkg/collector/python/memory.go [hunk 2]
```diff
@@ -69,9 +68,12 @@ func MemoryTracker(ptr unsafe.Pointer, sz C.size_t, op C.rtloader_mem_ops_t) {
 		case C.DATADOG_AGENT_RTLOADER_FREE:
 			bytes, ok := pointerCache.Load(ptr)
 			if !ok {
-				// log.Printf("untracked memory was attempted to be freed")
 				log.Warnf("untracked memory was attempted to be freed")
+				log.Warnf("Risky, printing pointer C-string: %s\n", C.GoString((*C.char)(ptr)))
```
[the remark this answers is not in the document]
Comment (truthbk) [reply pkg/collector/python/memory.go]: I'm removing this log line and cleaning up, I had it for debugging purposes but forgot cleaning up, we should definitely not try to print the untracked pointer!!!

Other than that, on the idea is that crashing here would imply you're doing something wrong on the allocation side. Kind of what would happen if you double freed, or called free on a pointer you did not own, etc. So I figured that was an acceptable behavior - if you weren't going to crash here, you were probably going to crash when the code got to the actual `free()`.

**Review by ogaca-dd — APPROVED** [review]

**Comment by truthbk** [comment]
There seems to be some untracked memory on windows (but code is stable) - we'll fix on a separate PR.

**Pull request merged**